src/state/sagas/runner/ListAllRunners/ListAllRunners.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+import { all, call, put, select, takeEvery } from 'redux-saga/effects';
+import { Api } from '../../../../services/config/Api';
+import { RUNNER_RUN_STATE } from '../../../../services/config/ApiConstants';
+import { ACL_PERMISSIONS } from '../../../../services/config/accessControl';
+import { ApiUtils, RunnersUtils } from '../../../../utils';
+import { STATUSES } from '../../../commons/Constants';
+import { RUNNER_ACTIONS_KEY } from '../../../commons/RunnerConstants';
+
+const getUserEmail = (state) => state.auth.userEmail;
+const getUserId = (state) => state.auth.userId;
+const getRunnersPermissionsMapping = (state) => state.application.permissionsMapping.runner;
+const getSolutionParameters = (state) => state.solution?.current?.data?.parameters;
+const getSolutionRunTemplates = (state) => state.solution?.current?.data?.runTemplates;
+
+const keepOnlyReadableRunners = (runners) =>
+  runners.filter((runner) => runner.security.currentUserPermissions.includes(ACL_PERMISSIONS.RUNNER.READ));
+
+function* getRunnerStatus(organizationId, workspaceId, runnerId, lastRunId) {
+  const response = yield call(Api.RunnerRuns.getRunStatus, organizationId, workspaceId, runnerId, lastRunId);
+  yield put({ type: RUNNER_ACTIONS_KEY.UPDATE_RUNNER, data: { runnerState: response.data.state, runnerId } });
+}
+export function* listAllRunners(organizationId, workspaceId) {
+  const userEmail = yield select(getUserEmail);
+  const userId = yield select(getUserId);
+  const runTemplates = yield select(getSolutionRunTemplates);
+  const runnersPermissionsMapping = yield select(getRunnersPermissionsMapping);
+  const solutionParameters = yield select(getSolutionParameters);
+  const { data } = yield call(Api.Runners.listRunners, organizationId, workspaceId);
```

Comment:
As discussed, this endpoint uses pagination, we need to force a very large page size to make sure we receive all runners :slightly_smiling_face: